modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-evaluator-impl/src/main/java/com/liferay/dynamic/data/mapping/form/evaluator/internal/helper/DDMFormEvaluatorHelper.java

```diff
@@ -595,24 +597,31 @@ private boolean _isFieldEmpty(
 			DDMFormFieldValue localizedObjectFieldDDMFormFieldValue =
 				new DDMFormFieldValue();
 
+			localizedObjectFieldDDMFormFieldValue.setDDMFormValues(
+				new DDMFormValues(ddmFormField.getDDMForm()));
 			localizedObjectFieldDDMFormFieldValue.setName(
 				ddmFormFieldValue.getName());
 
 			String defaultLanguageId = null;
+			String defaultLocale = String.valueOf(
+				ddmFormField.getProperty("defaultLocale"));
 
-			try {
-				JSONObject defaultLocaleJSONObject =
-					JSONFactoryUtil.createJSONObject(
-						String.valueOf(
-							ddmFormField.getProperty("defaultLocale")));
+			if (JSONUtil.isJSONObject(defaultLocale)) {
+				try {
+					JSONObject defaultLocaleJSONObject =
+						JSONFactoryUtil.createJSONObject(defaultLocale);
 
-				defaultLanguageId = defaultLocaleJSONObject.getString(
-					"localeId");
-			}
-			catch (JSONException jsonException) {
-				_log.error(jsonException);
+					defaultLanguageId = defaultLocaleJSONObject.getString(
+						"localeId");
+				}
+				catch (JSONException jsonException) {
+					_log.error(jsonException);
 
-				return true;
+					return true;
+				}
+			}
+			else {
+				defaultLanguageId = defaultLocale;
```

Comment:
What about if we did this:

```
String defaultLanguageId = String.valueOf(
	ddmFormField.getProperty("defaultLocale"));

if (JSONUtil.isJSONObject(defaultLocale)) {
	try {
		JSONObject defaultLocaleJSONObject =
			JSONFactoryUtil.createJSONObject(defaultLanguageId);

		defaultLanguageId = defaultLocaleJSONObject.getString(
			"localeId");
	}
	catch (JSONException jsonException) {
		_log.error(jsonException);

		return true;
	}
}
```

With this organization, I think we need to rename:
```
String defaultLanguageId = String.valueOf(
	ddmFormField.getProperty("defaultLocale"));
```
to
```
String defaultLocale = String.valueOf(
	ddmFormField.getProperty("defaultLocale"));
```
What do you think?

Reply:
Done!